prov/tcp/src/tcpx_comm.c

```diff
@@ -57,6 +57,27 @@ int tcpx_send_msg(struct tcpx_pe_entry *pe_entry)
 	return FI_SUCCESS;
 }
 
+static int is_pe_entry_size_matched(struct tcpx_pe_entry *pe_entry)
+{
+	uint64_t rx_xfer_size, recv_buf_size;
+
+	rx_xfer_size = (ntohll(pe_entry->msg_hdr.size) -
+			sizeof(pe_entry->msg_hdr));
+	recv_buf_size = ofi_total_iov_len(pe_entry->msg_data.iov,
+					  pe_entry->msg_data.iov_cnt);
+
+	if (rx_xfer_size == recv_buf_size)
+		return FI_SUCCESS;
+
+	if (rx_xfer_size > recv_buf_size)
+		return -FI_EINVAL;
+
+	ofi_truncate_iov(pe_entry->msg_data.iov,
+			 &pe_entry->msg_data.iov_cnt,
+			 rx_xfer_size);
```

Comment:
The name of the function implies that it simply returns true or false.  But we're modifying the pe entry.  We could optimize this by removing this function and calling ofi_truncate_iov() directly.  This would require modifying ofi_convert_iov() back to returning a status value to indicate if truncation occurred (success), or if the iov was too small (fi_etrunc).

Reply:
The msg interface is FIFO based, so always selecting the first entry is correct.  If the buffer overruns, that should be reported as an error to the user.